src/dtoannotations/Validator.java

```diff
@@ -12,55 +14,44 @@
-                        emailValueChars.add(fieldValue.toString().charAt(i));
-                    }
-                    for (int j = 0; j < regex.size(); j++) {
-                        if (!emailValueChars.get(j).equals(regex.get(j))) {
-                            list.add("\"" + declaredField.getName() + "\" " + "Field has an error, " +
-                                    "its value contains unacceptable symbol - "
-                                    + emailValueChars.get(j));
-                            break;
-                        }
-                    }
+            if (declaredField.isAnnotationPresent(Email.class) && fieldValue != null) {
+                Email email = declaredField.getAnnotation(Email.class);
+                Pattern pattern = Pattern.compile(email.emailRegex());
+                Matcher matcher = pattern.matcher(fieldValue.toString());
+                if (!matcher.matches()) {
+                        list.add("\"" + declaredField.getName() + "\" " + "Field has an error, " +
+                                "its value contains unacceptable symbol");
                 }
             }
-            if (declaredField.isAnnotationPresent(Adulthood.class)) {
-                if (fieldValue != null) {
-                    LocalDate minAge = LocalDate.of(2001, 1, 1);
-                    LocalDate maxAge = LocalDate.of(1960, 1, 1);
-                    LocalDate age = (LocalDate) fieldValue;
-                    if (age.isBefore(minAge) || age.isAfter(maxAge)) {
-                        list.add("\"" + declaredField.getName() + "\" " + "Field has an error, " +
-                                "age is - " + age);
-                    }
+            if (declaredField.isAnnotationPresent(Adulthood.class) && fieldValue != null) {
+                int currentYear = LocalDate.now().getYear();
+                int year = currentYear - ((LocalDate) fieldValue).getYear();
+                if (year < 18) {
+                    list.add("\"" + declaredField.getName() + "\" " + "Field has an error, " +
+                            "age is - " + year);
                 }
             }
-            if (declaredField.isAnnotationPresent(Min.class) && declaredField.isAnnotationPresent(Max.class)) {
+            if (declaredField.isAnnotationPresent(Min.class) && fieldValue != null) {
                 Min min = declaredField.getAnnotation(Min.class);
-                Max max = declaredField.getAnnotation(Max.class);
-                if (fieldValue != null) {
-                    if ((int) fieldValue < min.minValue() || (int) fieldValue > max.maxValue()) {
+                    if ((int) fieldValue < min.minValue()) {
                         list.add("\"" + declaredField.getName() + "\" " + "Field has an error, " +
-                                "discount can't be - " + fieldValue);
+                                "discount can't be less than " + fieldValue);
```

Comment:
Here must be the value from the **Annotation** Min(**20**). So change **fieldValue** into **min.minValue();**